src/theory/uf/theory_uf.h

```diff
@@ -162,7 +162,8 @@ class TheoryUF : public Theory {
                     const TNodeTrie* t2,
                     unsigned arity,
                     unsigned depth);
-
+  /** Is t a higher order type? */
```

Comment:
Say that this is checking explicitly for tn being a function type and the criterion is one of the arguments having a function type.